staging/src/k8s.io/apimachinery/pkg/watch/watch.go

```diff
@@ -165,11 +165,20 @@ type RaceFreeFakeWatcher struct {
 	result  chan Event
 	Stopped bool
 	sync.Mutex
+	extraErr bool
 }
 
 func NewRaceFreeFake() *RaceFreeFakeWatcher {
 	return &RaceFreeFakeWatcher{
-		result: make(chan Event, DefaultChanSize),
+		result:   make(chan Event, DefaultChanSize),
+		extraErr: false,
+	}
+}
+
+func NewRaceFreeFakeWithExtraError() *RaceFreeFakeWatcher {
```

Comment:
Function `NewRaceFreeFakeWithExtraError()` is not needed.  Method `RaceFreeFakeWatcher.Error()` already allows an error to be sent to the watcher.  Use that method to test with errors.  Please see https://github.com/kubernetes/kubernetes/blob/a4d0a03905b1a97f86d6db2fc17d882c6b526986/staging/src/k8s.io/apimachinery/pkg/watch/watch.go#L262

Reply:
The sending of Error depends on the new flag in RaceFreeFakeWatcher.
Without the flag, did you imply the following ?
```
diff --git a/staging/src/k8s.io/apimachinery/pkg/watch/watch.go b/staging/src/k8s.io/apimachinery/pkg/watch/watch.go
index 3945be3ae6..235f45f9f2 100644
--- a/staging/src/k8s.io/apimachinery/pkg/watch/watch.go
+++ b/staging/src/k8s.io/apimachinery/pkg/watch/watch.go
@@ -225,6 +225,7 @@ func (f *RaceFreeFakeWatcher) Modify(obj runtime.Object) {
        if !f.Stopped {
                select {
                case f.result <- Event{Modified, obj}:
+                       f.Error(nil)
                        return
                default:
                        panic(fmt.Errorf("channel full"))
```

Reply:
@vladimirvivien 
See if the above snippet aligns with your comment.